kairon/shared/utils.py

```diff
@@ -1701,3 +1704,46 @@ def validate_faq_training_file(faq_file: File):
         required_headers = {'questions', 'answer'}
         if not required_headers.issubset(columns):
             raise AppException(f"Required columns {required_headers} not present in file.")
+
+
+class StoryValidator:
+
+    @staticmethod
+    def get_graph(steps: List) -> DiGraph:
+        graph = DiGraph()
+        vertices = {}
+        for story_step in steps:
+            vertices[story_step['step']['name']] = KaironStoryStep(story_step["step"]["name"],
+                                                                   story_step["step"]["type"])
+        for story_step in steps:
+            story_step_object = vertices[story_step["step"]["name"]]
+            for connected_story_step in story_step['connections'] or []:
+                connection_object = vertices[connected_story_step["name"]]
+                graph.add_edge(story_step_object, connection_object)
+        return graph
+
+    @staticmethod
+    def validate_steps(steps: List):
```

Comment:
add another validation: intent can only have one connection of action type.
otherwise intent predictions will fail